src/ui/public/state_management/state_hashing/unhash_query_string.js

```diff
@@ -0,0 +1,8 @@
+import { mapValues } from 'lodash';
+
+export default function unhashQueryString(parsedQueryString, states) {
```

Comment:
I see your point. We can write the code in a certain way that encourages developers to use the "right name" when importing the module.

On the other hand, I think the pattern of using default exports to indicate a single export from a module is also pretty common, especially for single-function modules like these.  I think a practice like "alias the default export with same name as the file" is a pretty straightforward practice to adopt, and will address the problem of developers naming the import poorly.

The AirBnB style guide recommends [preferring a default export with modules with only one export](https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/prefer-default-export.md), though they don't do a very good job of explaining why. I don't think we should adhere to it for the sake of dogma, but it's worth some thought as to why they recommend this practice. [Bullet 3.2 of this guide](http://www.2ality.com/2014/09/es6-modules-final.html) has a pretty good possible explanation.